### ISSUE TITLE
Add Player.retryRequest()

### ISSUE THREAD

**Issue by rkuroiwa** [google/shaka-player#1772]
**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**
Yes.

**Is your feature request related to a problem? Please describe.**

This is a follow up on https://github.com/google/shaka-player/issues/1771
If shaka player signaled the app that there are no keys available, then the app can decide to take an action. 

**Describe the solution you'd like**

`Player.retryRequest()` will restart from creating a new `MediaKeySession`, from the same `MediaKeys` instance that is attached to the media element, and generates a request (`MediaKeySession.generateRequest()`).
For "persistent-license" sessions, use `MediaKeySession.load()` instead.

Basically it restarts from loading a license into the CDM.

**Describe alternatives you've considered**

If it is going to be a configuration rather than an notification to the app, this doesn't need to be exposed, but the functionality is still required.

**Additional context**

This may be also useful for other cases. For example, if the user decides to put their computer to sleep for the duration of the license, and when they wake up the computer, all the keys are "expired". The app may decide to call `Player.retryRequest()` to "renew" the license.

**Comment by joeyparrish** [comment]
I would suggest a different name: `player.retryLicensing()`.  Retrying a "request" is very generic, and the name could apply to all kinds of network requests.  Since we want something that parallels `player.retryStreaming()`, but for DRM licenses, I think `retryLicensing()` is the better name.

I'll rename the issue to reflect that.

**Comment by cristian-atehortua** [comment]
This looks like a nice feature. On our side, we are facing an issue on Tizen devices when the user leaves the app with the player opened. When he comes back after some minutes, we ran into an error because the player tries to resume playback but as the license expired and it was not renewed since the app was in the background, Shaka just crash.

Even better, is it possible Shaka automatically tries to renew the license when it was not able to renew because the app is in the background?

**Comment by tykus160** [comment]
New methods:
- `shaka.Player#retryLicensing(sessionMetadata: shaka.extern.DrmSessionMetadata, retryDelaySeconds = 0.1): boolean` to retrigger license request for given session metadata. Metadata can be retrieved i.e. from `shaka.Player#getActiveSessionsMetadata()`

New configs:
- `drm.failureCallback: function(error: shaka.util.Error): void` no-op by default. Called on DRM Engine errors, i.e. on `LICENSE_REQUEST_FAILED`, so it can be used to manually retry license request. `error.handled` should be set to `true` in a callback to prevent throwing fatal error

Modifications:
- `LICENSE_REQUEST_FAILED` should contain `shaka.extern.DrmSessionMetadata` of failed request

**Comment by cristian-atehortua** [comment]
Hi, thanks for the update 

May this retry function work on LICENSE_EXPIRED errors?

**Comment by tykus160** [comment]
Hey @cristian-atehortua 
For now my main goal is to request a new license either periodically without an explicit reason, or on `LICENSE_REQUEST_FAILED` error. I also took a bit another approach than originally suggested, so we don't create new MediaKeySessions, but generate new requests on existing ones. That's why `DrmSessionMetadata` needs to be provided to `retryLicensing()`.

`EXPIRED` handling would be a nice addition, but I won't cover it on initial PR at least. Main problem is we fire this error only when all sessions are already expired. I think in order to make it work in workflow I provided, we would have to throw this error on any expired session.

**Comment by tykus160** [comment]
My assumptions here were incorrect and it cannot be implemented in a way I described. `MediaKeySession` allows to call `generateRequest()` only once - if session is initialized, it will fail, see [here](https://w3c.github.io/encrypted-media/#dom-mediakeysession-generaterequest).

Due to that, on `retryLicensing()` we should just close all existing sessions and try to create & open new ones.

Open questions:
1. Will video element throw an error during this process? Can we prevent that?
2. Will new encrypted event fire? If not, will it limit this feature on some platforms?

**Comment by avelad** [comment]
@KimKyuHoi do you want work on it after https://github.com/shaka-project/shaka-player/pull/9589?

**Comment by KimKyuHoi** [comment]
> @KimKyuHoi do you want work on it after https://github.com/shaka-project/shaka-player/pull/9589?

Sounds good. I'll start working on it as soon as #9589 is finished.

**Comment by avelad** [comment]
@KimKyuHoi now it's possible to start with this, your PR has been merged.
